server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java

```diff
@@ -372,4 +444,14 @@ protected String getMasterActionName(DiscoveryNode node) {
         return getClusterManagerActionName(node);
     }
 
+    /**
+     * Determines if transport action needs to check local cluster-state term with manager before
+     * executing the action on manager. This is generally true for actions that are read-only and can be executed locally
+     * on node if the term matches with cluster-manager.
+     * @return - true to perform term check and then execute the action
+     */
+    protected boolean checkTermVersion() {
```

Comment:
This seems like an anti-pattern to and this property for a single case.  I would rather we use interfaces to describe different transport action 'properties', what do you think?

Reply:
I am starting with cluster/state but this will be extended for other transport actions of ClusterManager.

Reply:
@peternied - i am thinking to have the sub-classes override the method on need basis here (one to start-with and possibility of multiple others in future). There does not seem to be that much functionality to extract out as interface hierarchy. Let me know your thoughts.